File: smart_chicken_farming/lib/model/ThongBao.dart
class Thongbao {
  final int id;
  final String tieu_de;
  final String noi_dung;
  final String loai_tb;
  bool da_doc;
  final DateTime thoi_gian_tao;

  Thongbao({
    required this.id,
    required this.tieu_de,
    required this.noi_dung,
    required this.loai_tb,
    required this.da_doc,
    required this.thoi_gian_tao,
  });

  factory Thongbao.fromJson(Map<String, dynamic> json) {
    return Thongbao(
      id: json['id'] as int,
      tieu_de: json['tieu_de'] as String,
      noi_dung: json['noi_dung'] as String,
      loai_tb: json['loai_tb'] as String,
      da_doc: json['da_doc'] as bool,
      thoi_gian_tao: DateTime.parse(json['thoi_gian_tao'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tieu_de': tieu_de,
      'noi_dung': noi_dung,
      'loai_tb': loai_tb,
      'da_doc': da_doc,
      'thoi_gian_tao': thoi_gian_tao.toIso8601String(),
    };
  }
}

File: backend/src/main/java/com/example/backend/entities/ThongBao.java
package com.example.backend.entities;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "THONG_BAO")
@Data
public class ThongBao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String tieu_de;
    private String noi_dung;
    private String loai_tb;//thông báo trường hợp khẩn cấp hay thông báo nhắc nhỏ công tác chăn nuôi
    private boolean da_doc;
    private LocalDateTime thoi_gian_tao;

    @ManyToOne
    @JoinColumn(name = "id_lua_ga")
    @JsonIgnore
    private LuaGa luaGa;
}